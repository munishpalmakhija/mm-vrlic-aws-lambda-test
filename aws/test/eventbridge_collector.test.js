/*
Copyright 2018 VMware, Inc.
SPDX-License-Identifier: MIT
*/
const nock = require('nock');
const { lintTestEnv } = require('./helper.test');
const { eventBridgeSampleData } = require('./sample_files/eventbridge_testdata');

const {
  EventBridgeHttpCollector,
} = require('../src/index');


describe('EventBridgeHttpCollector', () => {
  const collector = new EventBridgeHttpCollector(lintTestEnv);

  describe('processLogsJson', () => {
    it('should process logs JSON', () => {
      const logsJson = eventBridgeSampleData();
      collector.processLogsJson(logsJson);
      expect(logsJson).toMatchSnapshot();
    });
  });
});

