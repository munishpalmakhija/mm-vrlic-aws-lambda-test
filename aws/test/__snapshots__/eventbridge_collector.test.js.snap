// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventBridgeHttpCollector processLogsJson should process logs JSON 1`] = `
Object {
  "account": "827066587692",
  "detail": Object {
    "currentState": "QUEUED",
    "queryExecutionId": "f11ed4a0-7c17-4172-a04f-56f79646dbe9",
    "sequenceNumber": "1",
    "statementType": "DDL",
    "versionId": "0",
    "workgroupName": "primary",
  },
  "detail-type": "Athena Query State Change",
  "id": "8908e14b-ec95-cca1-c478-57ea5929f5e8",
  "ingest_timestamp": 1538769915817,
  "log_type": "aws_eventbridge",
  "region": "us-east-2",
  "resources": Array [],
  "source": "aws.athena",
  "time": "2020-09-23T07:01:44Z",
  "version": "0",
}
`;
